src/repositories/versionable/VersionableRepository.ts

```diff
@@ -6,116 +6,104 @@ import { idText } from 'typescript';
 // import { isNull } from 'util';
 
 export default class VersionableRepository<I extends Document, M extends Model<I>> {
-  private model: M;
-  constructor(model) {
-    this.model = model;
-  }
-  /**
-   * @description Generate Object ID
-   * @returns objectID
-   */
-  protected static generateObjectId() {
-    return String(new Types.ObjectId());
-  }
+	private model: M;
+	constructor(model) {
+		this.model = model;
+	}
+	/**
+	 * @description Generate Object ID
+	 * @returns objectID
+	 */
+	protected static generateObjectId() {
+		return String(new Types.ObjectId());
+	}
 
-  /**
-   *
-   * @param query
-   * @returns one Document
-   */
-  protected findOne(query: any): Query<EnforceDocument<I, {}>, EnforceDocument<I, {}>, {}, I> {
-    console.log('Query for findOne in versionable::', query);
-    const finalQuery = { deletedAt: undefined, ...query };
-    // const findOneData = this.model.findOne(finalQuery).lean();
-    // console.log('in versionRepo- final', finalQuery);
-    // console.log('in versionRepo- final', findOneData);
-    return this.model.findOne(finalQuery).lean();
-  }
+	/**
+	 *
+	 * @param query
+	 * @returns one Document
+	 */
+	protected findOne(query: any): Query<EnforceDocument<I, {}>, EnforceDocument<I, {}>, {}, I> {
+		console.log('Query for findOne in versionable::', query);
+		const finalQuery = { deletedAt: undefined, ...query };
+		return this.model.findOne(finalQuery).lean();
+	}
 
-  /**
-   * @description Find documents
-   * @param query
-   * @param projection
-   * @param options
-   * @returns Array of Documents
-   */
-  protected find(
-    query,
-    projection?: any,
-    options?: any
-  ): Query<EnforceDocument<I, {}>[], EnforceDocument<I, {}>> {
-    const { skip = 0, limit = 10, sortBy = '-createdAt' } = query;
-    const finalQuery = { deletedAt: undefined, ...query };
-    const findData = this.model.find(finalQuery, projection, options);
-    console.log(findData);
-    return this.model
-      .find(finalQuery, projection, { skip: +skip, limit: +limit })
-      .sort(`-${sortBy}`);
-  }
+	/**
+	 * @description Find documents
+	 * @param query
+	 * @param projection
+	 * @param options
+	 * @returns Array of Documents
+	 */
+	protected find(query, projection?: any, options?: any): Query<EnforceDocument<I, {}>[], EnforceDocument<I, {}>> {
+		const { skip = 0, limit = 10, sortBy = '-createdAt' } = query;
+		const finalQuery = { deletedAt: undefined, ...query };
+		const findData = this.model.find(finalQuery, projection, options);
+		console.log(findData);
+		return this.model.find(finalQuery, projection, { skip: +skip, limit: +limit }).sort(`-${sortBy}`);
```

Comment:
what is this +skip and +limit

Reply:
What does + signify here?

Reply:
its doing nothing just a showcase of should be positive and number only.